src/main/java/com/theelixrlabs/healthcare/service/DoctorPatientAssignmentService.java

```diff
@@ -0,0 +1,86 @@
+package com.theelixrlabs.healthcare.service;
+
+import com.theelixrlabs.healthcare.constants.DoctorPatientAssignmentConstants;
+import com.theelixrlabs.healthcare.dao.DoctorPatientAssignmentRepository;
+import com.theelixrlabs.healthcare.dto.DoctorPatientAssignmentDto;
+import com.theelixrlabs.healthcare.exceptionHandler.CustomException;
+import com.theelixrlabs.healthcare.model.DoctorPatientAssignmentModel;
+import com.theelixrlabs.healthcare.repository.DoctorRepository;
+import com.theelixrlabs.healthcare.repository.PatientRepository;
+import com.theelixrlabs.healthcare.utility.MessageUtil;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class DoctorPatientAssignmentService {
+
+    private final DoctorPatientAssignmentRepository doctorPatientAssignmentRepository;
+    private final PatientRepository patientRepository;
+    private final DoctorRepository doctorRepository;
+    private final MessageUtil messageUtil;
+
+    public DoctorPatientAssignmentService(DoctorPatientAssignmentRepository doctorPatientAssignmentRepository,
+                                          PatientRepository patientRepository, DoctorRepository doctorRepository, MessageUtil messageUtil) {
+        this.doctorPatientAssignmentRepository = doctorPatientAssignmentRepository;
+        this.patientRepository = patientRepository;
+        this.doctorRepository = doctorRepository;
+        this.messageUtil = messageUtil;
+    }
+
+    private Optional<DoctorPatientAssignmentModel> validateDoctorPatientAssignment(DoctorPatientAssignmentDto doctorPatientAssignmentDTO) throws CustomException {
+        UUID doctorId, patientId;
+        try {
+            doctorId = UUID.fromString(doctorPatientAssignmentDTO.getDoctorId());
+        } catch (IllegalArgumentException illegalArgumentException) {
+            throw new CustomException(messageUtil.getMessage(DoctorPatientAssignmentConstants.DOCTOR_ID_INVALID_KEY));
+        }
+        try {
+            patientId = UUID.fromString(doctorPatientAssignmentDTO.getPatientId());
+        } catch (IllegalArgumentException illegalArgumentException) {
+            throw new CustomException(messageUtil.getMessage(DoctorPatientAssignmentConstants.PATIENT_ID_INVALID_KEY));
+        }
+        if (!doctorRepository.existsById(doctorId))
```

Comment:
Add open close brackets for if conditions even if it's single line block.

Reply:
Resolved